homeassistant/components/plex/server.py

```diff
@@ -78,13 +89,73 @@ def _connect_with_url():
 
         _connect_with_url()
 
-    def clients(self):
-        """Pass through clients call to plexapi."""
-        return self._plex_server.clients()
-
-    def sessions(self):
-        """Pass through sessions call to plexapi."""
-        return self._plex_server.sessions()
+    def refresh_entity(self, machine_identifier, device, session):
+        """Forward refresh dispatch to media_player."""
+        dispatcher_send(
+            self._hass,
+            PLEX_UPDATE_MEDIA_PLAYER_SIGNAL.format(machine_identifier),
+            device,
+            session,
+        )
+
+    def update_platforms(self):
+        """Update the platform entities."""
+        available_clients = {}
+        new_clients = []
+
+        try:
+            devices = self._plex_server.clients()
+            sessions = self._plex_server.sessions()
+        except plexapi.exceptions.BadRequest:
+            _LOGGER.exception("Error requesting Plex client data from server")
+            return
+        except requests.exceptions.RequestException as ex:
+            _LOGGER.warning(
+                "Could not connect to Plex server: %s (%s)", self.friendly_name, ex
+            )
+            return
+
+        for device in devices:
+            available_clients[device.machineIdentifier] = {"device": device}
+
+            if device.machineIdentifier not in self._known_clients:
+                new_clients.append(device.machineIdentifier)
+                _LOGGER.debug("New device: %s", device.machineIdentifier)
+
+        for session in sessions:
+            for player in session.players:
+                available_clients.setdefault(
+                    player.machineIdentifier, {"device": player}
+                )
+                available_clients[player.machineIdentifier]["session"] = session
+
+                if player.machineIdentifier not in self._known_clients:
+                    new_clients.append(player.machineIdentifier)
+                    _LOGGER.debug("New session: %s", player.machineIdentifier)
+
+        new_entity_configs = []
+        for client_id, client_data in available_clients.items():
+            if client_id in new_clients:
+                new_entity_configs.append(client_data)
+            else:
+                self.refresh_entity(
+                    client_id, client_data["device"], client_data.get("session")
+                )
+
+        self._known_clients.extend(new_clients)
+
+        idle_clients = [
```

Comment:
```suggestion
        idle_clients = self._known_clients.difference(available_clients)
```
`idle_clients` will be a set now.